modules/core/src/v2/coins/eos.ts

```diff
@@ -541,7 +542,7 @@ export class Eos extends BaseCoin {
         rpc: new NoopJsonRpc(),
         signatureProvider: new NoopSignatureProvider(),
         chainId: self.getChainId(),
-        textDecoder: new TextDecoder(),
+        textDecoder: new EosTextDecoder(),
         textEncoder: new TextEncoder(),
```

Comment:
seems pretty likely here that the textEncoder is also going to have the same issues? I'm assuming we're not running into an error because the encoder here isn't being used? but for consistency sake, we should should try to also make this a `stringEncoder` under the hood so that the encoder/decoder work as proper inverses of each other

Reply:
we can have a test covering the encoding and decoding phase with various tools

Reply:
@0xabrarhussain `StringEncoder` doesn't actually exist.

Looking at [TextEncoder](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder), its primarily used to transform utf8 strings into uint8arrays.

Our `EosTextDecoder` is implemented to do the opposite - data back to utf8 strings.

Reply:
oh wow this behavior is pretty confusing, can we have some comments on it to make sure people won't misuse it?

Reply:
also what is TestDecoder doing? cause we are swapping it with EosTextDecoder, 

I'm guessing 
TestDecoder - uint8arrays into utf8 strings 
EosTextDecoder - uint8arrays into utf8 strings too?

Reply:
@shihao-guo In short, yes, we want `TextDecoder` and `EosTextDecoder` to produce identical results.

`TextDecoder` is a standard class provided in javascript. However, its implementation leads to crashes in OVC (underlying problem is in Electron, the framework used to package OVC into an executable).

To get around this issue, we'll be switching to Node's `StringDecoder` instead. But since eosjs expects a TextDecoder, we have to subclass it.

Reply:
can we try to retrieve encoder as `util.TextEncoder` ? or does that also result in type errors

Reply:
@0xabrarhussain this will lead to type errors. We would have to do something similar - subclass TextEncoder and override `encode` to use `util.TextEncoder`.